flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1211,6 +1211,8 @@ private void completePendingCheckpoint(PendingCheckpoint pendingCheckpoint)
         final long checkpointId = pendingCheckpoint.getCheckpointId();
```

Comment:
In the last commit message:
> For example, if a task sent state for the first time and then failed without completing checkpoint and restarting JM;

Do you mean JM failure? Or TM/task failure?
> For example, if a task sent state for the first time and then JM failed without completing checkpoint and restarting JM

?

Can you elaborate what is the scenario?

Reply:
I meant TM failure (will update the message on rebase).
 
I desribed the scenario in more detail in the ticket: https://issues.apache.org/jira/browse/FLINK-24611?focusedCommentId=17445399&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17445399

Reply:
can you replace the "For example" paragraph with more or less copy paste of this comment? and maybe comment in the code as well? 

Also regarding both commit message and comment:
> without completing checkpoint and restarting JM

maybe rephrase to

> without completing checkpoint and without restarting JM

? As it is, I was confused a bit.

Reply:
I've updated the commit message and commented in code (replaced with scenario from the ticket).
The `restarting JM` part is now gone.